core/modules/qmeta/QMeta.h

```diff
@@ -156,17 +160,53 @@ class QMeta  {
      *
      *  @param queryId:   Query ID, non-negative number.
      */
-    virtual void finishQuery(int queryId) = 0;
+    virtual void finishQuery(QueryId queryId) = 0;
 
     /**
-     *  @brief Find all queries currently executing.
+     *  @brief Generic interface for finding queries.
+     *
+     *  Returns set of query IDs which satisfy all selections specified in parameters.
+     *
+     *  Setting "completed" to true is equivalent to setting "status" to a set of
+     *  (COMPLETED, FAILED, ABORTED) but is based on different QInfo attribute,
+     *  it uses "completed" instead of "status". Similarly setting "completed"
+     *  to false is equivalent to setting "status" to (EXECUTING).
```

Comment:
"completed" and "returned" are "ternary booleans" which accept 0, 1, or -1 (or false, true and "unknown" if you wish). We do not have enum definition for query processing stages, instead QInfo class defines methods which return non-zero timestamp when corresponding stage is complete.
I could extend QInfo with the idea of processing stage and define new enum for it, though I'm not too comfortable with that. If we start passing vectors of enums or bitfield to this method things may become interesting when the set of possible stages/enums changes (same is true for status enums). I'll try to think how to make that safer, any ideas welcome.